### PR TITLE
2517 spring boot upgrade

### PR DESCRIPTION
**Pull request by ConorH994**
https://tools.hmcts.net/jira/browse/AM-2639

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java [hunk 1]
```diff
@@ -1,75 +1,57 @@
...
-import uk.gov.hmcts.reform.roleassignment.controller.endpoints.CreateAssignmentController;
-
-import java.util.List;
+import org.springframework.web.method.HandlerMethod;
 
 @Configuration
-@EnableSwagger2WebMvc
+@SecurityScheme(name = "Authorization", type = SecuritySchemeType.HTTP, bearerFormat = "JWT", scheme = "bearer")
+@SecurityScheme(name = "ServiceAuthorization", type = SecuritySchemeType.APIKEY,
+    in = SecuritySchemeIn.HEADER, bearerFormat = "JWT", description = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java:21]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java [hunk 2]
```diff
@@ -1,75 +1,57 @@
...
-        return new ParameterBuilder()
-            .name("ServiceAuthorization")
-            .description("Valid Service-to-Service JWT token for a whitelisted micro-service")
-            .modelRef(new ModelRef(VALUE))
-            .parameterType(HEADER)
-            .required(true)
+    public GroupedOpenApi publicApi(@Autowired OperationCustomizer customGlobalHeaders) {
+        return GroupedOpenApi.builder()
+            .group("am-role-assignment-service")
+            .pathsToMatch("/**")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java:32]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/CreateAssignmentController.java [hunk 3]
```diff
@@ -43,28 +43,31 @@ public CreateAssignmentController(@Autowired CreateRoleAssignmentOrchestrator cr
         produces = V1.MediaType.CREATE_ASSIGNMENTS,
         consumes = {"application/json"}
     )
+    @Operation(summary = "Creates role assignments",
+        security =
+            {
+                @SecurityRequirement(name = "Authorization"),
+                @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/CreateAssignmentController.java:50]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/CreateAssignmentController.java [hunk 4]
```diff
@@ -43,28 +43,31 @@ public CreateAssignmentController(@Autowired CreateRoleAssignmentOrchestrator cr
...
+        description = V1.Error.INVALID_ROLE_NAME
+    )
+    @ApiResponse(
+        responseCode = "400",
+        description = V1.Error.INVALID_REQUEST
+    )
+    @ApiResponse(
+        responseCode = "422",
+        description = V1.Error.UNPROCESSABLE_ENTITY_REQUEST_REJECTED
+    )
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/CreateAssignmentController.java:70]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java [hunk 5]
```diff
@@ -94,26 +93,29 @@ public ResponseEntity<Void> deleteRoleAssignment(@RequestHeader(value = "x-corre
...
+        description = "No Content"
+    )
+    @ApiResponse(
+        responseCode = "400",
+        description = V1.Error.BAD_REQUEST_INVALID_PARAMETER
+    )
+    @ApiResponse(
+        responseCode = "400",
+        description = V1.Error.BAD_REQUEST_MISSING_PARAMETERS
+    )
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java:114]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java [hunk 6]
```diff
@@ -52,27 +49,29 @@ public DeleteAssignmentController(@Autowired DeleteRoleAssignmentOrchestrator de
         path = "am/role-assignments",
         produces = V1.MediaType.DELETE_ASSIGNMENTS
     )
+    @Operation(summary = "Delete role assignments",
+        security =
+            {
+                @SecurityRequirement(name = "Authorization"),
+                @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java:56]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java [hunk 7]
```diff
@@ -94,26 +93,29 @@ public ResponseEntity<Void> deleteRoleAssignment(@RequestHeader(value = "x-corre
         path = "am/role-assignments/{assignmentId}",
         produces = V1.MediaType.DELETE_ASSIGNMENTS
     )
+    @Operation(summary = "Delete role assignments by assignment Id",
+        security =
+            {
+                @SecurityRequirement(name = "Authorization"),
+                @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/DeleteAssignmentController.java:100]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/QueryAssignmentController.java [hunk 8]
```diff
@@ -49,19 +49,22 @@ public ResponseEntity<Object> getIdLdDemo(@RequestHeader(value = "x-correlation-
...
-        ),
-        @ApiResponse(
-            code = 400,
-            message = V1.Error.BAD_REQUEST_INVALID_PARAMETER
-        ),
+    @Operation(summary = "Query role assignments",
+        security =
+            {
+                @SecurityRequirement(name = "Authorization"),
+                @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/QueryAssignmentController.java:56]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java [hunk 9]
```diff
@@ -1,75 +1,57 @@
...
+            Parameter serviceAuthorizationHeader = new Parameter()
+                .in(ParameterIn.HEADER.toString())
+                .schema(new StringSchema())
+                .name("ServiceAuthorization")
+                .description("service-to-service token for a whitelisted micro-service")
+                .required(true);
+            Parameter authorizationHeader = new Parameter()
+                .in(ParameterIn.HEADER.toString())
+                .schema(new StringSchema())
+                .name("Authorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/config/SwaggerConfiguration.java:48]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/GetAssignmentController.java [hunk 10]
```diff
@@ -78,15 +81,18 @@ public ResponseEntity<RoleAssignmentResource> retrieveRoleAssignmentsByActorId(
         path = "/am/role-assignments/roles",
         produces = V1.MediaType.GET_ROLES
     )
+    @Operation(summary = "Get roles",
+        security =
+            {
+                @SecurityRequirement(name = "Authorization"),
+                @SecurityRequirement(name = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/controller/endpoints/GetAssignmentController.java:88]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/util/Swagger2SpringBoot.java [hunk 11]
```diff
@@ -1,14 +1,18 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.context.annotation.ComponentScan;
-import springfox.documentation.swagger2.annotations.EnableSwagger2WebMvc;
 
 @SpringBootApplication
-@EnableSwagger2WebMvc
+@SecurityScheme(name = "Authorization", type = SecuritySchemeType.HTTP, bearerFormat = "JWT", scheme = "bearer")
+@SecurityScheme(name = "ServiceAuthorization", type = SecuritySchemeType.APIKEY,
+    in = SecuritySchemeIn.HEADER, bearerFormat = "JWT", description = "ServiceAuthorization")
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/util/Swagger2SpringBoot.java:15]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f.

NB: class couldn;t be removed wihtout breaking integration tests.  So one to review in future ticket.

File: src/main/resources/application.yaml [hunk 12]
```diff
@@ -169,8 +174,6 @@ org :
   request:
     byPassOrgDroolRule: ${BYPASS_ORG_DROOL_RULE:false}
 
-logging.level.uk.gov.hmcts: ${APPLICATION_LOGGING_LEVEL:INFO}
```
[the remark this answers is not in the document]
Comment (mattnayler) [reply src/main/resources/application.yaml:172]: Resolved in #1896 see bc92c6d87e9d21d925f89871f9f25cf45d3b7e9f

File: src/main/java/uk/gov/hmcts/reform/roleassignment/oidc/JwtGrantedAuthoritiesConverter.java [hunk 13]
```diff
@@ -51,7 +52,7 @@ private List<GrantedAuthority> extractAuthorityFromClaims(List<String> roles) {
         return roles.stream()
             .map(SimpleGrantedAuthority::new)
             .map(GrantedAuthority.class::cast)
-            .toList();
+            .collect(Collectors.toList());
```
Comment (mattnayler) [inline src/main/java/uk/gov/hmcts/reform/roleassignment/oidc/JwtGrantedAuthoritiesConverter.java:55]: See [sonar report](https://sonarcloud.io/summary/new_code?id=am-role-assignment-service&pullRequest=1785)

Comment (mikebrownccd) [reply src/main/java/uk/gov/hmcts/reform/roleassignment/oidc/JwtGrantedAuthoritiesConverter.java]: collect(Collectors.toList()) changed back to toList() . I think this was only change originally to fix a Sonarqube issue and now Sonarqube is complaining wanting it changed back!

**Review by mattnayler — APPROVED** [review]

**Pull request merged**